test/bigtable/filter.js

```diff
@@ -0,0 +1,677 @@
+        startKeyExclusive: start.value,
+        endKeyExclusive: end.value
+      });
+    });
+  });
+
+  describe('parse', function() {
+    it('should call each individual filter method', function() {
+      var row = Filter.prototype.row;
+      var value = Filter.prototype.value;
+
+      Filter.prototype.row = spy();
+      Filter.prototype.value = spy();
+
+      var fakeFilter = [{
+        row: 'a'
+      }, {
+        value: 'b'
+      }];
+
+      Filter.parse(fakeFilter);
+
+      assert.strictEqual(Filter.prototype.row.calls.length, 1);
+      assert.strictEqual(Filter.prototype.row.calls[0].args[0], 'a');
+      assert.strictEqual(Filter.prototype.value.calls.length, 1);
+      assert.strictEqual(Filter.prototype.value.calls[0].args[0], 'b');
+
+      Filter.prototype.row = row;
+      Filter.prototype.value = value;
+    });
+
+    it('should throw an error for unknown filters', function() {
+      var fakeFilter = [{
+        wat: 'a'
+      }];
+
+      assert.throws(
+        Filter.parse.bind(null, fakeFilter),
+        /Unknown filter\: wat/
+      );
+    });
+
+    it('should return the filter in JSON form', function() {
+      var toProto = Filter.prototype.toProto;
+      var fakeProto = { a: 'a' };
+
+      Filter.prototype.toProto = spy(function() {
+        return fakeProto;
+      });
+
+      var fakeFilter = [{
+        column: 'a'
+      }];
+
+      var parsedFilter = Filter.parse(fakeFilter);
+
+      assert.strictEqual(parsedFilter, fakeProto);
+      assert.strictEqual(Filter.prototype.toProto.calls.length, 1);
+
+      Filter.prototype.toProto = toProto;
+    });
+  });
+
+  describe('all', function() {
+    it('should create a pass all filter when set to true', function(done) {
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'passAllFilter');
+        assert.strictEqual(value, true);
+        done();
+      };
+
+      filter.all(true);
+    });
+
+    it('should create a block all filter when set to false', function(done) {
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'blockAllFilter');
+        assert.strictEqual(value, true);
+        done();
+      };
+
+      filter.all(false);
+    });
+  });
+
+  describe('column', function() {
+    it('should set the column qualifier regex filter', function(done) {
+      var columnName = 'fake-column';
+      var convert = Filter.convertToRegExpString;
+
+      Filter.convertToRegExpString = spy(function(value) {
+        return value;
+      });
+
+      filter.set = function(filterName, value) {
+        assert.strictEqual(filterName, 'columnQualifierRegexFilter');
+        assert.strictEqual(value, columnName);
+        assert.strictEqual(
+          Filter.convertToRegExpString.calls[0].args[0], columnName);
+        assert.strictEqual(
+          FakeMutation.convertToBytes.calls[0].args[0], columnName);
+
+        Filter.convertToRegExpString = convert;
```

Comment:
I think if you move this up before the asserts, if the asserts fail, the override will still be reversed in time for future tests to use the real `convertToRegExpString`.